File: src/main/java/net/unaussprechlich/project/connect/gui/NotificationGUI.java
package net.unaussprechlich.project.connect.gui;

import net.minecraftforge.client.event.ClientChatReceivedEvent;
import net.minecraftforge.client.event.GuiOpenEvent;
import net.unaussprechlich.managedgui.lib.elements.GUI;
import net.unaussprechlich.managedgui.lib.elements.defaults.container.DefNotificationContainer;
import net.unaussprechlich.managedgui.lib.event.EnumDefaultEvents;
import net.unaussprechlich.managedgui.lib.event.util.EnumTime;
import net.unaussprechlich.managedgui.lib.event.util.Event;
import net.unaussprechlich.managedgui.lib.handler.MouseHandler;

import java.util.ArrayList;
import java.util.stream.Collectors;

/**
 * NotificationGUI Created by Alexander on 23.02.2017.
 * Description:
 **/
public class NotificationGUI extends GUI{

    private static final int SPACING = 1;

    private ArrayList<DefNotificationContainer> notifications = new ArrayList<>();

    public void addNotification(DefNotificationContainer notify){
        notifications.add(notify);
        registerChild(notify);
        updatePositions();
    }

    private void updatePositions(){
        int yPos = 0;
        for(DefNotificationContainer notify : notifications){
            notify.setYOffset(yPos);
            yPos += notify.getHeight() + SPACING;
        }
    }

    @Override
    public boolean doClientTick() {
        return true;
    }

    @Override
    public boolean doRender(int xStart, int yStart) {
        return true;
    }

    @Override
    public boolean doChatMessage(ClientChatReceivedEvent e) {
        return false;
    }

    @Override
    public boolean doMouseMove(int mX, int mY) {
        return false;
    }

    @Override
    public boolean doScroll(int i) {
        return false;
    }

    @Override
    public boolean doClick(MouseHandler.ClickType clickType) {
        return false;
    }

    @Override
    public <T extends Event> boolean doEventBus(T event) {
        if(event.getID() == EnumDefaultEvents.TIME.get() && event.getData() == EnumTime.SEC_1){
            notifications.removeAll(notifications.stream().filter(cont -> cont.getShowtime_sec() <= 0).collect(Collectors.toList()));
            updatePositions();
        }
        return true;
    }

    @Override
    public boolean doOpenGUI(GuiOpenEvent e) {
        return true;
    }
}

File: src/main/java/net/unaussprechlich/project/connect/gui/NotificationGUI.kt
package net.unaussprechlich.project.connect.gui

import net.minecraftforge.client.event.ClientChatReceivedEvent
import net.minecraftforge.client.event.GuiOpenEvent
import net.unaussprechlich.managedgui.lib.event.EnumDefaultEvents
import net.unaussprechlich.managedgui.lib.event.util.EnumTime
import net.unaussprechlich.managedgui.lib.event.util.Event
import net.unaussprechlich.managedgui.lib.gui.GUI
import net.unaussprechlich.managedgui.lib.handler.MouseHandler
import net.unaussprechlich.managedgui.lib.templates.defaults.container.DefNotificationContainer
import java.util.*

/**
 * NotificationGUI Created by Alexander on 23.02.2017.
 * Description:
 */
object NotificationGUI : GUI() {

    private val notifications = ArrayList<DefNotificationContainer>()

    fun addNotification(notify: DefNotificationContainer) {
        notifications.add(notify)
        registerChild(notify)
        updatePositions()
    }

    private fun updatePositions() {
        var yPos = 0
        for (notify in notifications) {
            notify.yOffset = yPos
            yPos += notify.height + SPACING
        }
    }

    override fun doClientTick(): Boolean {
        return true
    }

    override fun doRender(xStart: Int, yStart: Int): Boolean {
        return true
    }

    override fun doChatMessage(e: ClientChatReceivedEvent): Boolean {
        return false
    }

    override fun doMouseMove(mX: Int, mY: Int): Boolean {
        return false
    }

    override fun doScroll(i: Int): Boolean {
        return false
    }

    override fun doClick(clickType: MouseHandler.ClickType): Boolean {
        return false
    }

    override fun <T : Event<*>> doEventBus(event: T): Boolean {
        if (event.id == EnumDefaultEvents.TIME.get() && event.data === EnumTime.SEC_1) {
            notifications.removeAll(notifications.filter({ cont -> cont.showtime_sec <= 0 }))
            updatePositions()
        }
        return true
    }

    override fun doOpenGUI(e: GuiOpenEvent): Boolean {
        return true
    }

    override fun doResize(): Boolean {
        return true
    }

    private val SPACING = 1

}
